### PR TITLE
[Fix] Fix mmcls for cifar10 config on dev-1.x

### PR DESCRIPTION
**Pull request by hanrui1sensetime**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

deployment for mmcls model failed when use cifar10 config.

## Modification

keep `PackClsInput` as the last of pipeline.

## BC-breaking (Optional)

None.

## Use cases (Optional)
```bash
python tools/deploy.py configs/mmcls/classification_onnxruntime_dynamic.py resnet18_8xb16_cifar10.py resnet18_b16x8_cifar10_20210528-bd6371c8.pth tests/data/tiger.jpeg --work-dir mmdeploy_models/mmcls/ort --device cpu
```

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by AllentDan** [comment]
Pls rebase to dev-1.x

**Comment by hanrui1sensetime** [comment]
> Pls rebase to dev-1.x

Done.

File: mmdeploy/codebase/mmcls/deploy/classification.py [hunk 1]
```diff
@@ -180,6 +180,13 @@ def create_input(
         else:
             if pipeline[0]['type'] == 'LoadImageFromFile':
                 pipeline.pop(0)
+        # move PackClsInputs to the last of list for CIFAR10 dataset configs.
+        move_pipeline = []
+        while pipeline[-1]['type'] != 'PackClsInputs':
+            sub_pipeline = pipeline.pop(-1)
+            move_pipeline = [sub_pipeline] + move_pipeline
+        pipeline = pipeline[:-1] + move_pipeline + pipeline[-1:]
+
```
Comment (AllentDan) [inline mmdeploy/codebase/mmcls/deploy/classification.py:189]: I guess all you need is:
```
pipeline.append(pipeline.pop([item['type'] == 'PackClsInputs' for item in pipeline].index(True)))
```

**Review by AllentDan — APPROVED** [review]

**Comment by lvhan028** [comment]
converting to onnx is successful.
But when I run 
```
python demo/python/image_classification.py cpu mmdeploy_models/mmcls/ort tests/data/tiger.jpeg
```
It throws exception:
```
OpenCV Error: Sizes of input arguments do not match (The operation is neither 'array op array' (where arrays have the same size and the same number of channels), nor 'array op scalar', nor 'scalar op array') in arithm_op, file /build/opencv-L2vuMj/opencv-3.2.0+dfsg/modules/core/src/arithm.cpp, line 659
[2022-09-16 13:59:58.833] [mmdeploy] [error] [module_adapter.h:34] unhandled exception: /build/opencv-L2vuMj/opencv-3.2.0+dfsg/modules/core/src/arithm.cpp:659: error: (-209) The operation is neither 'array op array' (where arrays have the same size and the same number of channels), nor 'array op scalar', nor 'scalar op array' in function arithm_op

已放弃 (核心已转储)

```

**Comment by hanrui1sensetime** [comment]
> converting to onnx is successful. But when I run
> 
> ```
> python demo/python/image_classification.py cpu mmdeploy_models/mmcls/ort tests/data/tiger.jpeg
> ```
> 
> It throws exception:
> 
> ```
> OpenCV Error: Sizes of input arguments do not match (The operation is neither 'array op array' (where arrays have the same size and the same number of channels), nor 'array op scalar', nor 'scalar op array') in arithm_op, file /build/opencv-L2vuMj/opencv-3.2.0+dfsg/modules/core/src/arithm.cpp, line 659
> [2022-09-16 13:59:58.833] [mmdeploy] [error] [module_adapter.h:34] unhandled exception: /build/opencv-L2vuMj/opencv-3.2.0+dfsg/modules/core/src/arithm.cpp:659: error: (-209) The operation is neither 'array op array' (where arrays have the same size and the same number of channels), nor 'array op scalar', nor 'scalar op array' in function arithm_op
> 
> 已放弃 (核心已转储)
> ```

Done.

File: mmdeploy/backend/sdk/export_info.py [hunk 2]
```diff
@@ -212,6 +213,12 @@ def get_preprocess(deploy_cfg: mmengine.Config, model_cfg: mmengine.Config):
         item for item in transforms if 'Random' not in item['type']
         and 'RescaleToZeroOne' not in item['type']
     ]
+    # move PackClsInputs to the last of list for CIFAR10 dataset configs.
+    move_pipeline = []
+    while transforms[-1]['type'] not in PACK_LIST:
```
Comment (AllentDan) [inline mmdeploy/backend/sdk/export_info.py:218]: is regular expression better for this case?

**Comment by lvhan028** [comment]
May merge branch dev-1.x

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**